### PR TITLE
Execute filtered tests with single XCTest invocation

### PR DESCRIPTION
**Pull request by egorzhdan**
This improves performance by decreasing the number of spawned XCTest processes & makes the testing results output more convenient – `Executed 100 tests, with 0 failures` will be printed instead of `Executed 1 tests, with 0 failures` 100 times.

This is the next step after https://github.com/apple/swift-corelibs-xctest/pull/268

### CODE REVIEW

**Comment by egorzhdan** [comment]
Hey @aciidb0mb3r, could you please take a look? I'll resolve the conflicts if this PR makes sense in general.

File: Tests/CommandsTests/TestToolTests.swift [hunk 1]
```diff
@@ -30,6 +31,37 @@ final class TestToolTests: XCTestCase {
...
+                result = stderr
+            }
+            XCTAssert(result.contains("Test Suite 'ParallelTestsFailureTests' failed"))
+            XCTAssert(result.contains("Test Suite 'ParallelTestsTests' passed"))
+            XCTAssert(result.contains("Executed 3 tests, with 1 failure"))
+        }
+        #endif
+    }
+
+    func testFilteredTestCase() throws {
```
Comment (aciidb0mb3r) [inline Tests/CommandsTests/TestToolTests.swift]: We probably already have such integration tests, can you update the existing ones as these are costly to run?

**Comment by aciidb0mb3r** [comment]
Ooh, I see you added support for this in corelibs-xctest 😃

Please go ahead and rebase!

File: Sources/Commands/SwiftTestTool.swift [hunk 2]
```diff
@@ -231,21 +223,21 @@ public class SwiftTestTool: SwiftTool<TestToolOptions> {
...
-                        xctestArg: test.specifier,
-                        processSet: processSet,
-                        toolchain: toolchain,
-                        diagnostics: diagnostics,
-                        options: self.options,
-                        buildParameters: try self.buildParameters()
-                    )
-                    ranSuccessfully = runner.test() && ranSuccessfully
-                }
+                xctestArg = tests.map { $0.specifier }.joined(separator: ",")
```
[the remark this answers is not in the document]
Comment (egorzhdan) [reply Sources/Commands/SwiftTestTool.swift]: It does! There is a test for this in the XCTest project: https://github.com/apple/swift-corelibs-xctest/blob/e52fc6555b5aa9923f053d9a54660fa4b2c54690/Tests/Functional/SelectedTest/main.swift#L4

File: Tests/CommandsTests/TestToolTests.swift (hunk 1 above)
Comment (egorzhdan) [reply Tests/CommandsTests/TestToolTests.swift]: Oh right, I didn't notice the integration tests. There is currently a test which invokes `swift test` (`XCBuildTests.testSwiftTest`) but it seems to be specific to the Xcode build system, so I've added a new one – `BasicTests.testSwiftTest`.

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test

**Pull request merged**